lib/execution_engine2/utils/SDKMethodRunner.py

```diff
@@ -117,6 +143,20 @@ def run_job(self, params, user_id):
         params["service_ver"] = git_commit_hash
 
         # insert initial job document
-        job_id = self._init_job_rec(user_id, params)
+        job_id = self._init_job_rec(ctx["user_id"], params)
+
+        logging.info("About to run job with")
+        logging.info(params)
+        logging.info(ctx)
+        params["job_id"] = job_id
+        params["user_id"] = ctx["user_id"]
+        params["token"] = ctx["token"]
+        params["cg_resources_requirements"] = client_groups
+        try:
+            condor_job_id = self.condor.run_job(params)
```

Comment:
I think before we call `condor.run_job`, `params` should be processed already. `extract_resources`, `normalize` should happen in this class. Looks like this class processed `params` a little bit adding `job_id`, `token`, etc. And then `Condor` class processes client_groups further. But I can live with it for now. If it becomes more complicated, we might think about moving formatting `params` in one place.

Reply:
So I was struggling with where we would put that. The params could go to a different scheduler in the future, and they are mostly used for creating the condor submit file, so I thought they should be coupled with the condor utils.  Job_id/User_id/Token would probably go to running other schedulers as well.

Reply:
👍